### PR TITLE
Fix velocity distortion implementation

### PR DESCRIPTION
**Pull request by msz-rai**

### CODE REVIEW

File: src/gpu/RaytraceRequestContext.hpp [hunk 1]
```diff
@@ -20,16 +20,22 @@
 struct RaytraceRequestContext
 {
 	// Input
+	Vec6f* sensorVelocity;
+
```
Comment (prybicki) [inline src/gpu/RaytraceRequestContext.hpp:24]: 1. Why Vec6f? IMHO it is less obvious than 2xVec3f.
2. I'd use pass-by-value, same as rayOriginToWorld is passed.
```suggestion
	Vec3f sensorLinearVelocity;
	Vec3f sensorAngularVelocity;
```

File: src/gpu/optixPrograms.cu [hunk 2]
```diff
@@ -97,6 +103,21 @@ extern "C" __global__ void __raygen__()
 	const int rayIdx = optixGetLaunchIndex().x;
 	Mat3x4f ray = ctx.rays[rayIdx];
 
+	if (shouldDistort()) {
+		static const float toDeg = (180.0f / M_PI);
+		Vec3f linearVelocity((*ctx.sensorVelocity)[0], (*ctx.sensorVelocity)[1], (*ctx.sensorVelocity)[2]);
+		Vec3f angularVelocity((*ctx.sensorVelocity)[3], (*ctx.sensorVelocity)[4], (*ctx.sensorVelocity)[5]);
```
Comment (prybicki) [inline src/gpu/optixPrograms.cu:109]: See?

File: src/graph/NodesCore.hpp [hunk 3]
```diff
@@ -127,10 +127,14 @@ struct RaytraceNode : Node, IPointsNode
...
 	{ return std::const_pointer_cast<const VArray>(fieldData.at(field)); }
 
+	// RaytraceNode specific
+	void setVelocity(const Vec3f* linearVelocity, const Vec3f* angularVelocity);
+
 private:
 	std::shared_ptr<Scene> scene;
 	IRaysNode::Ptr raysNode;
 	VArrayProxy<Vec2f>::Ptr defaultRange = VArrayProxy<Vec2f>::create(1);
+	VArrayProxy<Vec6f>::Ptr sensorVelocity = VArrayProxy<Vec6f>::create();  // x, y, z, roll, pitch, yaw
```
Comment (prybicki) [inline src/graph/NodesCore.hpp:137]: You can name it angularVelocityRPY to fix interpretation.

Comment (prybicki) [reply src/graph/NodesCore.hpp:137]: Also, passing it by value removes VArray usage.

**Review by prybicki — APPROVED** [review]

File: src/gpu/optixPrograms.cu [hunk 4]
```diff
@@ -148,9 +161,17 @@ extern "C" __global__ void __closesthit__()
 		return;
 	}
 
+	// Fix XYZ if distortion is applied (XYZ must be calculated in sensor coordinate frame)
+	if (ctx.doApplyDistortion) {
+		const int rayIdx = optixGetLaunchIndex().x;
+		Mat3x4f undistortedRay = ctx.rays[rayIdx];
+		Vec3f undistortedOrigin = undistortedRay * Vec3f{0, 0, 0};
```
Comment (PiotrMrozik) [inline src/gpu/optixPrograms.cu:168]: This line tells clearly that we need a method in mat library.
sth like `Transform.ToUnit()`

**Review by PiotrMrozik — APPROVED** [review]
One small suggestion.
LGTM!

**Pull request merged**